src/coreclr/vm/methodtable.cpp

```diff
@@ -3338,23 +3349,33 @@ int MethodTable::GetLoongArch64PassStructInRegisterFlags(CORINFO_CLASS_HANDLE cl
                             size = STRUCT_FIRST_FIELD_DOUBLE;
                         }
                     }
+                    else if (pFieldStart[0].GetSize() == 8)
+                    {
+                        size = STRUCT_FIRST_FIELD_SIZE_IS8;
+                    }
                 }
                 else if (fieldType == ELEMENT_TYPE_CLASS)
                 {
-                    MethodTable* pMethodTable2  = pFieldStart->GetFieldTypeHandleThrowing().GetMethodTable();
+                    MethodTable* pMethodTable2;
 
-                    if (pMethodTable2->IsArray())
+                    TypeHandle th = pFieldStart->LookupFieldTypeHandle();
```

Comment:
`LookupFieldTypeHandle` can return null when the type was not loaded yet. I do not think it is correct to use it here.

Reply:
Yes, you are right.
I had met this case.

How about adding `if (pMethodTable2 != nullptr)`

Reply:
> How about adding if (pMethodTable2 != nullptr) ?

That does not work either. This method has to be deterministic, always return the same output for given input.

Reply:
> > How about adding if (pMethodTable2 != nullptr) ?
> 
> That does not work either. This method has to be deterministic, always return the same output for given input.

Thanks very much!

> there is another way that deleting all the `ELEMENT_TYPE_CLASS` which is being testing on my machine.

How about this ? I will update this PR liking this ?